models/twitter_user.go

```diff
@@ -0,0 +1,154 @@
+	URL  string `json:"url"`
+	Type string `json:"type"`
+
+	// User info
+	Name            string `json:"name"`
+	Username        string `json:"username"`
+	Bio             string `json:"bio"`
+	Avatar          string `json:"avatar"`
+	BackgroundImage string `json:"background_image"`
+
+	// Profile stats
+	Location   string `json:"location"`
+	JoinDate   string `json:"join_date"`
+	JoinTime   string `json:"join_time"`
+	IsPrivate  int    `json:"is_private"`
+	IsVerified int    `json:"is_verified"`
+
+	// Follows
+	Following     int      `json:"following"`
+	FollowingList []string `json:"following_list"`
+	Followers     int      `json:"followers"`
+	FollowersList []string `json:"followers_list"`
+
+	// Usage stats
+	Tweets     int `json:"tweets"`
+	Likes      int `json:"likes"`
+	MediaCount int `json:"media_count"`
+}
+
+// TwitterUser holds the follow graph info, only relating userNames
+type TwitterUser struct {
+	gorm.Model
+
+	// Meta
+	TwitterID string
+	URL       string
+	Type      string
+
+	// User info
+	Name            string
+	Username        string
+	Bio             string
+	Avatar          string
+	BackgroundImage string
+
+	// Profile stats
+	Location   string
+	JoinDate   time.Time
+	IsPrivate  bool
+	IsVerified bool
+
+	// Follows
+	Following     int
+	FollowingList []*TwitterUser `gorm:"many2many:twitter_followings;"`
+	Followers     int
+	FollowersList []*TwitterUser `gorm:"many2many:twitter_followers;"`
+
+	// Usage stats
+	Tweets     int
+	Likes      int
+	MediaCount int
+}
+
+// ConvertTwitterUser converts the raw TwitterUser structure
+// from kafka into the database model
+func ConvertTwitterUser(raw *TwitterUserRaw) *TwitterUser {
+
+	followingList := make([]*TwitterUser, len(raw.FollowingList))
+	followersList := make([]*TwitterUser, len(raw.FollowersList))
+
+	for index, item := range raw.FollowingList {
+		followingList[index] = &TwitterUser{
+			Username: item,
+		}
+	}
+
+	for index, item := range raw.FollowingList {
+		followingList[index] = &TwitterUser{
+			Username: item,
+		}
+	}
+
+	joinDate, _ := utils.ConvertDateStrToTime(raw.JoinDate)
+
+	isPrivate := utils.ConvertIntToBool(raw.IsPrivate)
+	isVerified := utils.ConvertIntToBool(raw.IsVerified)
+
+	return &TwitterUser{
+		TwitterID: raw.ID,
+		URL:       raw.URL,
+		Type:      raw.Type,
+
+		Name:            raw.Name,
+		Username:        raw.Username,
+		Bio:             raw.Bio,
+		Avatar:          raw.Avatar,
+		BackgroundImage: raw.BackgroundImage,
+
+		Location:   raw.Location,
+		JoinDate:   joinDate,
+		IsPrivate:  isPrivate,
+		IsVerified: isVerified,
+
+		Following:     raw.Following,
+		FollowingList: followingList,
+		Followers:     raw.Followers,
+		FollowersList: followersList,
+
+		Tweets:     raw.Tweets,
+		Likes:      raw.Likes,
+		MediaCount: raw.MediaCount,
+	}
+}
+
+// Create converts multiple TwitterUser slices into a single custom
+// TwitterUserList slice
+func (list *TwitterUserList) Create(slices ...[]*TwitterUser) {
```

Comment:
you can convert this to a  "constructor" func instead:
```
func NewTwitterUserList(slices ...[]*TwitterUser) TwitterUserList{
...
}
```
and do the remove duplicates inside the constructor (since you always call create and remove duplicates together). THat way you don't have to manage a pointer to a slice

Reply:
Up to you, how you handle this here. 
I personally prefer to keep the interface of something as simple/small as possible and only extend it, when it is actually needed without trying to anticipate the future use cases.